torch/_dynamo/variables/user_defined.py

```diff
@@ -244,6 +249,17 @@ def call_method(
                     args[0].as_python_constant() in self.value, **options
                 ).add_guard(self.source.make_guard(GuardBuilder.ODICT_KEYS))
 
+            if (
+                method is dict.__contains__
+                and len(args) == 1
+                and isinstance(args[0], (ConstantVariable, BuiltinVariable))
+                and inspect.getattr_static(type(self.value), "keys") is dict.keys
+            ):
+                assert not kwargs
+                return ConstantVariable(
+                    args[0].as_python_constant() in self.value, **options
+                ).add_guard(self.source.make_guard(GuardBuilder.ODICT_KEYS))
+
```

Comment:
L240-L262 are pretty similar logic for ```dict``` and ```OrderedDict```, can you consolidate them?

Reply:
Yep!